File: javaPrep/vsjava/LL/Dll.java
public class Dll {
    Node head;

    public class Node{
        Node pre;
        int data;
        Node next;

        public Node(int n){
            this.data = n;
            this.pre = null;
            this.next = null;
        }
    }

    public void addFirst(int n){
        if(head == null){
            Node newNode = new Node(n);
            head = newNode;
        }
        else{
            Node newNode = new Node(n);
            newNode.next = head;
            head.pre = newNode;
            head =head.pre;
        }
    }

    public void display(){
        while(head != null){
            System.out.print(head.data + " -> ");
            head = head.next;
        }
        System.out.println("null");
    }

    public void addLast(int n){
        if(head == null){
            addFirst(n);
        }
        else{
            Node newNode = new Node(n);
            Node temp = head;
            while(temp.next != null){
                temp = temp.next;
            }
            temp.next = newNode;
            newNode.pre = temp;
        }
    }

}

File: Cprep/linked List/dll.c
#include<stdio.h>
#include<stdlib.h>

struct node {
    struct node *pre;
    int data;
    struct node *nxt;

};

struct node * createNode(int value){
    struct node *newNode = malloc(sizeof(struct node));
    newNode->data = value;
    newNode->pre = NULL;
    newNode->nxt = NULL;
    return newNode;
}

struct node * addFirst(struct node *head, int value){
    if(head == NULL){
        head = createNode(value);
    }
    else{
        struct node *newNode = createNode(value);
        newNode->nxt = head;
        head->pre = newNode;
        head = head->pre;
    }
    return head;
}

struct node * addLast(struct node *head,int value){
    if(head == NULL){
        head = createNode(value);
    }
    else{
        struct node *newNode = createNode(value);
        struct node *temptr = head;
        while(temptr->nxt != NULL){
            temptr= temptr->nxt;
        }
        temptr->nxt = newNode;
        newNode->pre =  temptr;
    }
    return head;
}

struct node * addAtPos(struct node * head,int value){
    if(head == NULL){
        head = createNode(value);
    }
    else{
        struct node *newNode = createNode(value);
        

    }
}

void printList(struct node *head){
    while(head != NULL){
        printf("%d <-> ",head->data);
        head = head->nxt;
    }
    printf("NULL\n");
}




void main(){
    struct node * head = NULL;
    head = addFirst(head,21);
    head = addFirst(head,99);
    head = addFirst(head,65);
    head = addLast(head,0);
    head =addLast(head,11);
    printList(head);
}